llvm/include/llvm/Transforms/Vectorize/LoopVectorizationLegality.h

```diff
@@ -551,6 +602,17 @@ class LoopVectorizationLegality {
   /// (potentially) make a better decision on the maximum VF and enable
   /// the use of those function variants.
   bool VecCallVariantsFound = false;
+
+  /// Indicates whether this loop has a speculative early exit, i.e. an
+  /// uncountable exiting block that is not the latch.
+  bool HasSpeculativeEarlyExit = false;
+
+  /// Keep track of all the loop exiting blocks.
```

Comment:
Is this comment accurate? AFAICT they only keep track of a single countable and uncountable exiting block, *if* the backedge taken count is not computable.